test/iostream.jl

```diff
@@ -0,0 +1,19 @@
+@test mktemp() do _, file
+    skipchars(file, isspace)
+    true
+end
+
+@test eof(skipchars(IOBuffer("    "), isspace))
+@test eof(skipchars(IOBuffer("#    \n   "), isspace, linecomment='#'))
+
+macro test_skipchars(str, expected_char, lnc=Char(0xffffffff))
+  quote
```

Comment:
There's still a failure for trailing whitespace, though we should probably migrate that to a removal script that we just run once a week or so. We don't have an auto indentation checker that I know of, I just have an unusually sensitive eye for it.

[ci skip] does still work, but it's the weekend so things are quiet, no need for it at the moment.